langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/package.scala

```diff
@@ -144,102 +204,159 @@ package object semanticdb {
+      val dentries = sdocuments.documents.toIterator.map { sdoc =>
         try {
           toDb(sourcepath, sdoc)
         } catch {
           case NonFatal(e) =>
             throw new IllegalArgumentException(
-              s"Error converting s.Document to m.Document where filename=${sdoc.filename}\n$sdoc",
+              s"Error converting s.TextDocument to m.Document where uri=${sdoc.uri}\n$sdoc",
               e)
         }
       }
       d.Database(dentries.toList)
     }
   }
+
   implicit class XtensionDatabase(ddatabase: d.Database) {
-    def toSchema(sourceroot: AbsolutePath): s.Database = {
+    def toSchema(sourceroot: AbsolutePath): s.TextDocuments = {
       val sentries = ddatabase.documents.map {
         case d.Document(dinput, dlanguage, dnames, dmessages, dsymbols, dsynthetics) =>
           object dPosition {
-            def unapply(dpos: dPosition): Option[s.Position] = dpos match {
-              case org.langmeta.inputs.Position.Range(_, sstart, send) =>
-                Some(s.Position(sstart, send))
+            def unapply(dpos: dPosition): Option[s.Range] = dpos match {
+              case dpos: org.langmeta.Position.Range =>
+                Some(s.Range(dpos.startLine, dpos.startColumn, dpos.endLine, dpos.endColumn))
               case _ =>
                 None
             }
           }
+          object disDefinition {
+            def unapply(disDefinition: Boolean): Option[s.SymbolOccurrence.Role] = {
+              if (disDefinition) Some(s.SymbolOccurrence.Role.DEFINITION)
+              else Some(s.SymbolOccurrence.Role.REFERENCE)
+            }
+          }
           object dSeverity {
-            def unapply(dseverity: d.Severity): Option[s.Message.Severity] = {
+            def unapply(dseverity: d.Severity): Option[s.Diagnostic.Severity] = {
               dseverity match {
-                case d.Severity.Info => Some(s.Message.Severity.INFO)
-                case d.Severity.Warning => Some(s.Message.Severity.WARNING)
-                case d.Severity.Error => Some(s.Message.Severity.ERROR)
+                case d.Severity.Error => Some(s.Diagnostic.Severity.ERROR)
+                case d.Severity.Warning => Some(s.Diagnostic.Severity.WARNING)
+                case d.Severity.Info => Some(s.Diagnostic.Severity.INFORMATION)
+                case d.Severity.Hint => Some(s.Diagnostic.Severity.HINT)
                 case _ => None
               }
             }
           }
-          object dDenotation {
-            def unapply(ddefn: d.Denotation): Option[s.Denotation] = {
-              import ddefn._
-              val snames = ddefn.names.map {
-                case d.ResolvedName(org.langmeta.inputs.Position.Range(_, sstart, send), ssym, sisDefinition) =>
-                  s.ResolvedName(Some(s.Position(sstart, send)), ssym.syntax, sisDefinition)
-                case other =>
-                  sys.error(s"bad database: unsupported position $other")
+          object dResolvedSymbol {
```

Comment:
While were refactoring this, can we please add more expect tests for modifiers? The current expect tests don't handle all available modifiers, writing a single file like
```scala
package object App {
  trait Trait
  class Class
  object Object
  implicit def Implicit = 1
  sealed trait Sealed
  protected def Protected = 2
  def TypeParameter[T] = 3
  def Parameter(Param: Int) = 4
  ...
}
```

We have hit on roundtrip bugs and those are usually not discovered until we pull in the latest release in scalafix.

Reply:
Done.